File: 2405. Optimal Partition of String/string_4.cpp
# include <string>

class Solution {
public:
    int partitionString(string s) {
        int partitions = 1;
        string substring = "";
        for (char character: s){
            if (substring.find(character) != string::npos){
                partitions++;
                substring = "";
            }
            substring += character;
            //cout << substring << endl;
        }
        return partitions;
    }
};

File: 2405. Optimal Partition of String/string_4.py
class Solution:
    def partitionString(self, s: str) -> int:
        partitions = 1 # start with 1 as all characters might be unique
        substring = ''
        for character in s:
            if character in substring:
                partitions += 1
                substring = ''
            substring += character
        return partitions